ledger/sandbox/src/main/scala/com/digitalasset/platform/configuration/MetricsReporter.scala

```diff
@@ -52,8 +54,25 @@ object MetricsReporter {
       Graphite(new InetSocketAddress(defaultHost, port))
   }
 
-  implicit val metricsReporterRead: Read[MetricsReporter] = Read.reads {
-    _.split(":", 2).toSeq match {
+  implicit val metricsReporterRead: Read[MetricsReporter] = Read.reads { value =>
+    if (value == "console") Console
+    else if (!value.contains("://")) {
+      readLegacy(value)
+    } else {
+      val uri = Read.uriRead.reads(value)
+      uri.getScheme match {
+        case "csv" => Csv(Paths.get(uri.getPath))
+        case "graphite" =>
+          val port = if (uri.getPort > 0) uri.getPort else Graphite.defaultPort
+          val address = new InetSocketAddress(uri.getHost, port)
+          val metricPrefix = Some(uri.getPath.stripPrefix("/")).filter(_.nonEmpty)
+          Graphite(address, metricPrefix)
+      }
```

Comment:
Can you add a `case _` that throws a reasonable exception, like in the `readLegacy` function below?